ext/oss/sink.go

```diff
@@ -111,35 +122,49 @@ func (o *OSSSink) process() error {
 		// create file write handlers
 		wh, ok := o.writeHandlers[destinationURI]
 		if !ok {
-			// create new oss write handler
-			targetDestinationURI, err := url.Parse(destinationURI)
-			if err != nil {
-				o.Logger().Error(fmt.Sprintf("failed to parse destination URI: %s", destinationURI))
-				return errors.WithStack(err)
-			}
-			if targetDestinationURI.Scheme != "oss" {
-				o.Logger().Error(fmt.Sprintf("invalid scheme: %s", targetDestinationURI.Scheme))
-				return errors.WithStack(err)
-			}
-			// remove object if overwrite is enabled
-			if o.enableOverwrite {
-				o.Logger().Info(fmt.Sprintf("overwrite is enabled, remove object: %s", destinationURI))
-				if err := o.Retry(func() error {
-					err := o.remove(targetDestinationURI.Host, strings.TrimLeft(targetDestinationURI.Path, "/"))
-					return err
+			var oh io.Writer
+			if o.enableArchive {
+				tmpPath, err := getTmpPath(destinationURI)
+				if err != nil {
+					o.Logger().Error(fmt.Sprintf("failed to get tmp path for %s: %s", destinationURI, err.Error()))
+					return errors.WithStack(err)
+				}
+
+				oh, err = xio.NewWriteHandler(o.Logger(), tmpPath)
+				if err != nil {
+					o.Logger().Error(fmt.Sprintf("failed to create write handler: %s", err.Error()))
+					return errors.WithStack(err)
+				}
+			} else {
+				// create new oss write handler
+				targetDestinationURI, err := url.Parse(destinationURI)
+				if err != nil {
+					o.Logger().Error(fmt.Sprintf("failed to parse destination URI: %s", destinationURI))
+					return errors.WithStack(err)
+				}
+				if targetDestinationURI.Scheme != "oss" {
+					o.Logger().Error(fmt.Sprintf("invalid scheme: %s", targetDestinationURI.Scheme))
+					return errors.WithStack(err)
+				}
+				// remove object if overwrite is enabled
+				if o.enableOverwrite {
```

Comment:
to make the flow consistent, can we delete the oss object in upfront? regardless it's archive or not

Reply:
if the SDK supports removing all files having a specific prefix, we can do it